common/lib/protocol-definitions/src/protocol.ts

```diff
@@ -88,6 +88,11 @@ export enum SignalType {
      * System signal sent to indicate a client has left the collaboration.
      */
     ClientLeave = "leave",
+
+    /**
+     * System signal sent to indicate that the external source of truth for the data in the collaboration has changed.
+     */
+    ExternalDataChange = "externalDataChange",
```

Comment:
I do not think it's appropriate to add to this list.
It would be great to start with a document describing the design.
I'd think that we want to isolate micros-services / lambdas, i.e. I do not think it's appropriate for front-end (alfred) to be involved here. And if it's not alfred, then most likely signals need to be in "user" mode, i.e. generated by another FF client (even if such client lives in service).

It's also not very clear what are considerations around intermittency of connections and signals, and the fact that client may simply never receive this signal. Is that acceptable? Feels like you would want either ops (such that client at least can react once it connects / reconnects to the document).

Reply:
@vladsud this draft PR is to see if we can hack something together in order to write the design document. I have updated the body of the PR to reflect that. I am writing up the design document now.